app/helpers/duties_helper.rb

```diff
@@ -43,18 +43,63 @@ def process_day_place(duties)
   end
 
   def process_duty(duties)
```

Comment:
I don't want to see a 50 line method.

Reply:
Please break up this method into a few methods, or rewrite some parts of the code. Consult me if you have any problems.